azure-iot-hub-devicesdk/tests/test_authentication_provider_factory.py

```diff
@@ -6,14 +6,18 @@
 from azure.iot.hub.devicesdk.auth.authentication_provider_factory import (
     from_connection_string,
     from_shared_access_signature,
+    from_x509
 )
```

Comment:
out of scope: these names should be changed.  auth_provider = from_x509() sounds wrong.  It should be one of:
auth_provider = AuthorizationProvider.from_x509() 
or
auth_provider = authorization_provider_from_x509()